native/common/jp_boxedtype.cpp

```diff
@@ -46,7 +46,7 @@ JPMatch::Type JPBoxedType::findJavaConversion(JPMatch &match)
 	if (m_PrimitiveType->findJavaConversion(match) != JPMatch::_none)
 	{
 		JP_TRACE("Primitive", match.type);
-		match.conversion = boxConversion;
+		match.conversion = boxBooleanConversion;
```

Comment:
But it generates one weirdness here because we can't call the boxed conversion directly.

Reply:
Doesn't the conversion implement all abstract methods? What's the compilers message (if you wouldn't mind, I'm just curious).

Reply:
It gives the usual message about cannot create an instance of an object with a virtual method.  The problem is of course that some of the JConverters have match defined (when the convert will be something else, but we can derived in that case), some have convert defined (where the match rules require extra arguments that are special to the calling type), and most have both.  Something like...

```
native/common/jp_chartype.cpp:45:7: note:   because the following virtual functions are pure within ‘JPConversionAsChar’:
```

Reply:
I'm scratching my head right now, if we really need derivation and polymorphic lookups here. Why can't we just use templates?

But don't waste more time on this, it is almost perfect now.

Reply:
I managed to get templates for the Float and Long patterns.   But the Object converters are a mess because the match rules are often very complete.  I will merge this PR for now and take another run at making it cleaner after I have bit off a few of the bigger ones.